src/sentry/services/hybrid_cloud/auth/__init__.py

```diff
@@ -190,6 +197,27 @@ class AuthenticationResponse(AuthenticationContext):
     user_from_signed_request: bool = False
 
 
+@dataclass(eq=True, frozen=True)
+class ApiAuthProviderFlags:
+    allow_unlinked: bool = False
+    scim_enabled: bool = False
+
+
+@dataclass(eq=True, frozen=True)
+class ApiAuthProvider:
+    id: int = -1
+    organization_id: int = -1
+    provider: str = ""
+    flags: ApiAuthProviderFlags = field(default_factory=lambda: ApiAuthProviderFlags())
+
+
+@dataclass(eq=True)
+class ApiOrganizationAuthConfig:
```

Comment:
A batch group of auth related settings  that the organization (region silo) needs to know, but is configured via control silo models.